File: app/src/main/java/com/kent/university/privelt/ui/data/DataViewModel.java
package com.kent.university.privelt.ui.data;

import com.kent.university.privelt.model.Service;
import com.kent.university.privelt.model.UserData;
import com.kent.university.privelt.repositories.ServiceDataRepository;
import com.kent.university.privelt.repositories.UserDataRepository;

import java.util.List;
import java.util.concurrent.Executor;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

public class DataViewModel extends ViewModel {
    private final UserDataRepository mUserDataRepository;
    private final ServiceDataRepository mServiceDataRepository;
    private final Executor mExecutor;

    private LiveData<List<Service>> services;


    public DataViewModel(UserDataRepository mUserDataRepository, ServiceDataRepository mServiceDataRepository, Executor mExecutor) {
        this.mUserDataRepository = mUserDataRepository;
        this.mServiceDataRepository = mServiceDataRepository;
        this.mExecutor = mExecutor;
    }

    void init() {
        if (services == null)
            services = mServiceDataRepository.getServices();
    }

    LiveData<List<Service>> getServices() {
        return services;
    }

    LiveData<List<UserData>> getUserData(long id, String type) {
        return mUserDataRepository.getUserDatasForAServiceAndType(id, type);
    }
}

File: app/src/main/java/com/kent/university/privelt/ui/data/DataViewModel.kt
/*
 *  This Source Code Form is subject to the terms of the Mozilla Public
 *  * License, v. 2.0. If a copy of the MPL was not distributed with this
 *  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.kent.university.privelt.ui.data

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.kent.university.privelt.model.Service
import com.kent.university.privelt.model.UserData
import com.kent.university.privelt.repositories.ServiceDataRepository
import com.kent.university.privelt.repositories.UserDataRepository

class DataViewModel(private val mUserDataRepository: UserDataRepository, private val mServiceDataRepository: ServiceDataRepository) : ViewModel() {
    var services: LiveData<List<Service>>? = null
        private set

    fun init() {
        if (services == null) services = mServiceDataRepository.services
    }

    fun getUserData(id: Long, type: String?): LiveData<List<UserData>> {
        return mUserDataRepository.getUserDatasForAServiceAndType(id, type)
    }

}